pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -142,7 +142,6 @@ func readRuntimeObjectAsPipeline(ctx context.Context, namespace string, obj runt
 		// without actually creating the Pipeline on the cluster.
 		// Validation must happen before the v1beta1 Pipeline is converted into the storage version of the API,
 		// since validation of beta features differs between v1 and v1beta1
```

Comment:
This comment is no longer relevant and can be removed. I think we can also move this function call out of the switch statement.